src/main/java/ecommerce/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package ecommerce.dao.impl;
+
+import ecommerce.dao.OrderDao;
+import ecommerce.db.Storage;
+import ecommerce.lib.Dao;
+import ecommerce.model.Order;
+import ecommerce.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
```

Comment:
On my opinion on DAO layer we should take Order as argument to save, not ShoppingCart

Reply:
done, ty